docs/architecture.md

```diff
@@ -19,62 +19,62 @@
-- The Uplink UI is driven by the global `State` variable. It is initialized via `use_shared_state_provider`. We don't want this variable to be re-loaded every time the app updates, so `app_bootstrap` is used to initialize state. 
-- finally `app` is rendered. All the global channels are polled here, each with their own `use_future`. 
-    + conversations are loaded from warp and added to `State`
-    + friends are loaded from warp and added to `State`
+- in `main.rs`, there is a call to `dioxus_desktop::launch_cfg`. It is passed a `bootstrap` element, which is rendered first. 
+- Uplink needs to start `warp_runner` from within a `Tokio` runtime. That runtime isn't available until `dioxus_desktop::launch_cfg` is called. We don't want to restart `warp_runner` when the UI updates, so the `bootstrap` Element ensures this only happens once. 
+- next `auth_page_manager` is rendered. `Warp` requires a user account. Until an account is created, the user data can't be decrypted and loaded. `auth_page_manager` uses the `auth_wrapper` element to handle user authentication and account creation. When done, `app_bootstrap` is rendered. 
+- The global `State` variable drives the Uplink UI. It is initialized via `use_shared_state_provider`. We don't want this variable to be re-loaded every time the app updates, so `app_bootstrap` is used to initialize state. 
+- finally, `app` is rendered. All the global channels are polled here, each with their own `use_future`. 
+    + conversations are loaded from the warp and added to `State`
+    + friends are loaded from the warp and added to `State`
     + warp events are passed to `State::process_warp_event`
     + after all the `use_future`s, `cx.render()` is called with the various parts of the page.
 
 ## Global Variables
-- Uplink uses global variables to do some important things. They are all in `main.rs`. 
-- `STATIC_ARGS` is for constant variables which may be affected by command line options
+- Uplink uses global variables to do some essential things. They are all in `main.rs`. 
+- `STATIC_ARGS` is for constant variables that may be affected by command line options
 - the rest of the variables are channels. 
 
 ## Global Channels
-- `WARP_CMD_CH`: used to communicate with `warp_runner`. Rather than block the UI while `Warp` performs a task, `warp_runner` receives commands via the channel and responds via a oneshot channel. `WARP_CMD_CH.tx` is used from within a Dioxus Element, inside of a `use_future` or a `use_coroutine`. 
+- `WARP_CMD_CH`: used to communicate with `warp_runner`. Rather than block the UI while `Warp` performs a task, `warp_runner` receives commands via the channel and responds via a one-shot channel. `WARP_CMD_CH.tx` is used from within a Dioxus Element, inside a `use_future` or a `use_coroutine`. 
```

Comment:
```suggestion
- `WARP_CMD_CH`: used to communicate with `warp_runner`. Rather than block the UI while `Warp` performs a task, `warp_runner` receives commands via the channel and responds via a oneshot channel. `WARP_CMD_CH.tx` is used from within a Dioxus Element, inside a `use_future` or a `use_coroutine`. 
```